docs/deployment/cloudfoundry.md

```diff
@@ -0,0 +1,342 @@
+---
+id: cloudfoundry
+title: Deploying to Cloud Foundry
+sidebar_label: Cloud Foundry
+description: How to deploy Backstage to Cloud Foundry
+---
+
+Cloud Foundry is an open source cloud application platform that makes it faster and easier to build, test, deploy, and scale apps in your choice of cloud, framework, and language.
+
+## Prerequisites & the Cloud Foundry CLI
+
+This guide assumes a basic understanding of working on a cloud foundry environment specifically using the cloud foundry CLI or CF CLI for short. See the [getting started guide](https://docs.cloudfoundry.org/cf-cli/getting-started.html) for more information.
+
+## Getting Started
+
+Build your own developer portal. Follow the getting started guide on backstage to get started. https://backstage.io/docs/getting-started/
+
+Next, install the Cloud Foundry CLI and follow the instructions in the quickstart guide to login.
+
+Once we are situated with an instance of backstage and our cloud foundry CLI we can get started in configuring the backstage instance to work with Cloud Foundry.
+
+From the top level of your backstage project change directory into your backend packages folder. Here we will create our manifest.yml file as well as modify our package.json to accommodate our buildpack
+
+This is important because the nodejs buildpack see [buildpacks](https://docs.cloudfoundry.org/buildpacks/) does not support newer versions of yarn. We need to modify our package.json to swap any references of `link:..` With `file:..`
+Example to update:
+
+```
+"dependencies": {
+  ...
+  "@backstage/plugin-techdocs-backend": "^1.10.9",
+  "app": "link:../app",
+  ...
+},
+
+Example of updated packages/backend/package.json:
+
+"dependencies": {
+  ...
+  "@backstage/plugin-techdocs-backend": "^1.10.9",
+  "app": "file:../app",
+  ...
+},
+```
+
+### Building Backstage
+
+Build your backstage instance following the instructions below taken from https://backstage.io/docs/deployment/docker. Run the following command in the root directory of your backstage instance
+
+yarn build:backend --config ../../app-config.yaml
+
+Following this we want to take the built backstage and place it in a new build directory
+
+```
+cp -v \
+  yarn.lock  \
+  package.json  \
+  packages/backend/dist/skeleton.tar.gz  \
+  packages/backend/dist/bundle.tar.gz  \
+  app-config.yaml \
+  $BUILD_DIR
+```
+
+Since backstage uses yarn berry (> v1.x.x) and the buildpack for Cloud Foundry only supports v1.x.x and below we need to create a new .yarnrc file in the build directory
```

Comment:
I think some of these steps need to be updated as `@backstage/create-app` now automatically creates a `.yarnrc` file

Reply:
Hi @vinzscam, `@backstage/create-app` does generate a `.yarnrc.yml` file compatible with yarn 2+, but I didn't see a `.yarnrc` file in my run of create-app. In this case, creating a `.yarnrc` file is meant to support yarn v1.x

Reply:
ah makes sense, sorry for the confusion!

Reply:
Does the `.yarnrc.yml` file need to be manually removed?